// @react95/core/esm/ThemeProvider/themes/millenium.js
function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { _defineProperty(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import BaseTheme, { generateShadows } from './baseTheme';
var colors = {
  anchor: '#1034a6',
  anchorVisited: '#440381',
  borderDark: '#828282',
  borderDarkest: 'black',
  borderLight: '#e5e5e5',
  borderLighter: '#f0f0f0',
  borderLightest: '#f6f6f6',
  canvas: 'black',
  canvasText: '#ffffff',
  canvasTextDisabled: '#828282',
  canvasTextDisabledShadow: '#ffffff',
  canvasTextInvert: '#ffffff',
  headerBackground: 'linear-gradient(to right, #012470, #a5c7e7)',
  headerNotActiveBackground: '#7f7f7f',
  headerNotActiveText: '#d6cfc7',
  headerText: '#ffffff',
  material: '#d6cfc7',
  materialDark: '#9a9e9c',
  materialText: 'black',
  materialTextDisabled: '#828282',
  materialTextDisabledShadow: '#ffffff',
  materialTextInvert: '#ffffff',
  progress: '#00256e',
  inputBackground: '#ffffff',
  inputBackgroundDisabled: '#e5e5e5',
  tooltip: '#fefbcc'
};

var millenium = _objectSpread(_objectSpread({}, BaseTheme), {}, {
  colors: colors,
  shadows: generateShadows(colors)
});

export default millenium;
