// @react95/core/cjs/GlobalStyle/Borders.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.windowBorder = exports.scrollBarBorder = exports["default"] = exports.createBorder = exports.border = void 0;

var _styledComponents = require("styled-components");

var createBorder = function createBorder(_ref) {
  var direction = _ref.direction,
      outerTopLeft = _ref.outerTopLeft,
      innerTopLeft = _ref.innerTopLeft,
      outerBottomRight = _ref.outerBottomRight,
      innerBottomRight = _ref.innerBottomRight;
  return (0, _styledComponents.css)(["border:0;box-shadow:inset -1px -1px 0 0 ", ",inset 1px 1px 0 0 ", ",inset -2px -2px 0 0 ", ",inset 2px 2px 0 0 ", ";"], direction == 'extrude' ? outerBottomRight : innerTopLeft, direction == 'extrude' ? outerTopLeft : innerBottomRight, direction == 'extrude' ? innerBottomRight : outerTopLeft, direction == 'extrude' ? innerTopLeft : outerBottomRight);
};

exports.createBorder = createBorder;

var border = function border() {
  var _ref2 = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {},
      _ref2$direction = _ref2.direction,
      direction = _ref2$direction === void 0 ? 'extrude' : _ref2$direction;

  return createBorder({
    direction: direction,
    outerBottomRight: (0, _styledComponents.css)(["", ""], function (_ref3) {
      var theme = _ref3.theme;
      return theme.colors.borderDarkest;
    }),
    innerBottomRight: (0, _styledComponents.css)(["", ""], function (_ref4) {
      var theme = _ref4.theme;
      return theme.colors.borderDark;
    }),
    outerTopLeft: (0, _styledComponents.css)(["", ""], function (_ref5) {
      var theme = _ref5.theme;
      return theme.colors.borderLightest;
    }),
    innerTopLeft: (0, _styledComponents.css)(["", ""], function (_ref6) {
      var theme = _ref6.theme;
      return theme.colors.borderLight;
    })
  });
};

exports.border = border;

var windowBorder = function windowBorder() {
  var _ref7 = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {},
      _ref7$direction = _ref7.direction,
      direction = _ref7$direction === void 0 ? 'extrude' : _ref7$direction;

  return createBorder({
    direction: direction,
    outerBottomRight: (0, _styledComponents.css)(["", ""], function (_ref8) {
      var theme = _ref8.theme;
      return theme.colors.borderDarkest;
    }),
    innerBottomRight: (0, _styledComponents.css)(["", ""], function (_ref9) {
      var theme = _ref9.theme;
      return theme.colors.borderDark;
    }),
    outerTopLeft: (0, _styledComponents.css)(["", ""], function (_ref10) {
      var theme = _ref10.theme;
      return theme.colors.borderLight;
    }),
    innerTopLeft: (0, _styledComponents.css)(["", ""], function (_ref11) {
      var theme = _ref11.theme;
      return theme.colors.borderLightest;
    })
  });
};

exports.windowBorder = windowBorder;

var scrollBarBorder = function scrollBarBorder() {
  var _ref12 = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {},
      _ref12$direction = _ref12.direction,
      direction = _ref12$direction === void 0 ? 'extrude' : _ref12$direction;

  return createBorder({
    direction: direction,
    outerBottomRight: (0, _styledComponents.css)(["", ""], function (_ref13) {
      var theme = _ref13.theme;
      return theme.colors.borderDarkest;
    }),
    innerBottomRight: (0, _styledComponents.css)(["", ""], function (_ref14) {
      var theme = _ref14.theme;
      return theme.colors.borderDark;
    }),
    outerTopLeft: (0, _styledComponents.css)(["", ""], function (_ref15) {
      var theme = _ref15.theme;
      return theme.colors.borderLight;
    }),
    innerTopLeft: (0, _styledComponents.css)(["", ""], function (_ref16) {
      var theme = _ref16.theme;
      return theme.colors.borderLightest;
    })
  });
};

exports.scrollBarBorder = scrollBarBorder;
var _default = border;
exports["default"] = _default;
