// @react95/core/cjs/Tree/Node.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.icons = exports["default"] = void 0;

var React = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("@xstyled/styled-components"));

var _treeMid = _interopRequireDefault(require("./imgs/tree-mid.png"));

var _treeLast = _interopRequireDefault(require("./imgs/tree-last.png"));

var _treeNodeChildren = _interopRequireDefault(require("./imgs/tree-node-children.png"));

var _icons = require("@react95/icons");

var _excluded = ["children", "id", "icon", "label", "onClick"];

var _templateObject, _templateObject2, _templateObject3, _templateObject4, _templateObject5, _templateObject6;

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { var _i = arr == null ? null : typeof Symbol !== "undefined" && arr[Symbol.iterator] || arr["@@iterator"]; if (_i == null) return; var _arr = []; var _n = true; var _d = false; var _s, _e; try { for (_i = _i.call(arr); !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var icons = {
  FILE_MEDIA: _icons.MediaCd,
  FILE_TEXT: _icons.FileText,
  FILE_UNKNOWN: _icons.Bat,
  FILE_FONT: _icons.FileFont2,
  FILE_PEN: _icons.FilePen,
  FILE_SETTINGS: _icons.FileSettings,
  FILE_TEXT_SETTINGS: _icons.FileTextSettings,
  FILE_EXECUTABLE: _icons.BatExec
};
exports.icons = icons;

var NodeItem = _styledComponents["default"].div(_templateObject || (_templateObject = _taggedTemplateLiteral(["\n  list-style-type: none;\n  background-repeat: no-repeat;\n  background-image: url(", ");\n\n  &:last-child {\n    background-image: url(", ");\n  }\n"])), _treeMid["default"], function (_ref) {
  var isOpen = _ref.isOpen;
  return isOpen ? _treeMid["default"] : _treeLast["default"];
});

var NodeInfo = _styledComponents["default"].div(_templateObject2 || (_templateObject2 = _taggedTemplateLiteral(["\n  display: flex;\n  align-items: center;\n  user-select: none;\n  -webkit-user-select: none;\n  -khtml-user-select: none;\n  -moz-user-select: none;\n  -ms-user-select: none;\n"])));

var FolderStatus = _styledComponents["default"].div(_templateObject3 || (_templateObject3 = _taggedTemplateLiteral(["\n  display: flex;\n  align-items: center;\n  justify-content: center;\n  width: 10px;\n  height: 10px;\n  border: 1;\n  border-color: borderDarkest;\n  background-color: inputBackground;\n  font-size: 11px;\n"])));

var IconContainer = _styledComponents["default"].div(_templateObject4 || (_templateObject4 = _taggedTemplateLiteral(["\n  display: flex;\n  align-items: center;\n  justify-content: center;\n  width: 20px;\n  height: 20px;\n  margin-right: 6;\n  margin-left: ", "px;\n\n  > img {\n    width: 14px;\n    height: 14px;\n  }\n"])), function (_ref2) {
  var hasChildren = _ref2.hasChildren;
  return hasChildren ? 8 : 18;
});

var NodeChildren = _styledComponents["default"].ul(_templateObject5 || (_templateObject5 = _taggedTemplateLiteral(["\n  padding: 0 0 0 22;\n  background-image: url(", ");\n  background-repeat: repeat-y;\n"])), _treeNodeChildren["default"]);

var Label = _styledComponents["default"].span(_templateObject6 || (_templateObject6 = _taggedTemplateLiteral(["\n  outline: none;\n  padding: 1;\n\n  :focus {\n    border-width: 1;\n    border-style: dotted;\n    padding: 0;\n  }\n"])));

var NodeIcon = function NodeIcon(_ref3) {
  var hasChildren = _ref3.hasChildren,
      isOpen = _ref3.isOpen;

  if (!hasChildren) {
    return /*#__PURE__*/React.createElement(_icons.Bat, {
      variant: "32x32_4",
      "data-testid": "react95-default-icon-bat"
    });
  }

  if (isOpen) {
    return /*#__PURE__*/React.createElement(_icons.FolderOpen, {
      variant: "32x32_4",
      "data-testid": "react95-default-icon-folder-open"
    });
  }

  return /*#__PURE__*/React.createElement(_icons.Folder, {
    variant: "32x32_4",
    "data-testid": "react95-default-icon-folder"
  });
};

var Node = function Node(_ref4) {
  var _ref4$children = _ref4.children,
      children = _ref4$children === void 0 ? [] : _ref4$children,
      id = _ref4.id,
      icon = _ref4.icon,
      label = _ref4.label,
      _ref4$onClick = _ref4.onClick,
      onClick = _ref4$onClick === void 0 ? function () {} : _ref4$onClick,
      rest = _objectWithoutProperties(_ref4, _excluded);

  var _React$useState = React.useState(false),
      _React$useState2 = _slicedToArray(_React$useState, 2),
      isOpen = _React$useState2[0],
      setIsOpen = _React$useState2[1];

  var hasChildren = children.length > 0;

  var onClickHandler = function onClickHandler(event) {
    onClick(event, {
      id: id,
      icon: icon,
      label: label,
      children: children
    });
  };

  var onKeyDownHandler = function onKeyDownHandler(event) {
    if (event.key === ' ') {
      setIsOpen(!isOpen);
      onClickHandler(event);
    }
  };

  return /*#__PURE__*/React.createElement(NodeItem, _extends({
    isOpen: isOpen
  }, rest), /*#__PURE__*/React.createElement(NodeInfo, null, hasChildren && /*#__PURE__*/React.createElement(FolderStatus, {
    onClick: function onClick() {
      return setIsOpen(!isOpen);
    }
  }, isOpen ? '-' : '+'), /*#__PURE__*/React.createElement(IconContainer, {
    hasChildren: hasChildren
  }, icon || /*#__PURE__*/React.createElement(NodeIcon, {
    hasChildren: hasChildren,
    isOpen: isOpen
  })), /*#__PURE__*/React.createElement(Label, {
    tabIndex: 0,
    onDoubleClick: function onDoubleClick() {
      return setIsOpen(!isOpen);
    },
    onClick: onClickHandler,
    onKeyDown: onKeyDownHandler
  }, label)), hasChildren && isOpen && /*#__PURE__*/React.createElement(NodeChildren, null, children === null || children === void 0 ? void 0 : children.map(function (dataNode) {
    return /*#__PURE__*/React.createElement(Node, _extends({
      key: dataNode.id
    }, dataNode));
  })));
};

var _default = Node;
exports["default"] = _default;
